src/pytorch_lightning/utilities/upgrade_checkpoint.py

```diff
@@ -11,30 +11,81 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import argparse
+import glob
 import logging
+from argparse import ArgumentParser, Namespace
+from pathlib import Path
 from shutil import copyfile
+from typing import List
 
 import torch
+from tqdm import tqdm
 
 from pytorch_lightning.utilities.migration import migrate_checkpoint, pl_legacy_patch
 
-log = logging.getLogger(__name__)
+_log = logging.getLogger(__name__)
 
-if __name__ == "__main__":
 
-    parser = argparse.ArgumentParser(
+def _upgrade(args: Namespace) -> None:
+    path = Path(args.path).absolute()
+    extension: str = args.extension if args.extension.startswith(".") else f".{args.extension}"
+    files: List[Path] = []
+
+    if not path.exists():
+        _log.error(
+            f"The path {path} does not exist. Please provide a valid path to a checkpoint file or a directory"
+            f" containing checkpoints ending in {extension}."
+        )
+        exit(1)
+
+    if path.is_file():
+        files = [path]
+    if path.is_dir():
+        files = [Path(p) for p in glob.glob(str(path / "**" / f"*{extension}"), recursive=True)]
+    if not files:
+        _log.error(
+            f"No checkpoint files with extension {extension} were found in {path}."
+            f" HINT: Try setting the `--extension` option to specify the right file extension to look for."
+        )
+        exit(1)
+
+    _log.info("Creating a backup of the existing checkpoint files before overwriting in the upgrade process.")
+    for file in files:
+        backup_file = file.with_suffix(".bak")
+        if backup_file.exists():
+            # never overwrite backup files - they are the original, untouched checkpoints
+            continue
+        copyfile(file, backup_file)
+
+    _log.info("Upgrading checkpoints ...")
+    progress = tqdm if len(files) > 1 else lambda x: x
```

Comment:
Using tqdm can be useful even if there's just 1 file because then the user can see that there's just 1 file